include/aero/array/array.hpp

```diff
@@ -11,35 +14,41 @@ class Array {
   Array(const Array& other) = default;
   /// Move constructor
   Array(Array&& other) = default;
+  /// Constructor from dimensions. Array elements
+  /// are initialized to zero.
+  Array(const unsigned int number_of_elements);
+  /// Constructor from dimensions and initial value
+  Array(const unsigned int number_of_elements,
+      const Real initial_value);
+  /// Constructor from vector of reals
+  Array(const std::vector<Real> &values);
 
   /// Destructor
   virtual ~Array() {}
 
   /// Assignment operator
   Array& operator=(const Array&) = default;
+  /// Assignment from a vector of Reals
+  Array& operator=(const std::vector<Real> &values);
   /// Default move assignment operator
   Array& operator=(Array&&) = default;
-};
-
-/// This Array subclass allows access to arrays implemented in
-/// Fortran.
-class FortranArray: public Array {
-public:
-
-  /// Construct an aerosol model wrapped around a Fortran implementation
-  /// that can be accessed with the given pointer.
-  explicit FortranArray(void *fortran_pointer);
-
-  // Overridden functionality
-
-  FortranArray(FortranArray& other);
-  FortranArray(FortranArray&& other);
-  ~FortranArray() override;
 
-  FortranArray& operator=(FortranArray& other);
-  FortranArray& operator=(FortranArray&& other);
+  /// Returns a clone (deep-copy) of an Array
+  virtual Array* clone() const;
+  /// Copies data into an Array
+  virtual void copy_in(const Real *input);
+  /// Copies data into an Array
+  virtual void copy_in(const std::vector<Real> &input);
+  /// Copies data out of an Array
+  virtual void copy_out(Real *output) const;
+  /// Copies data out of an Array
+  virtual void copy_out(std::vector<Real> &output) const;
+  /// Returns the number of elements in an Array
+  virtual unsigned int size() const;
+
+private:
+  std::vector<Real> values_;
```

Comment:
I don't think we want to impose a representation on our `Array` type. For example, this would be useless for anything on a GPU. Is there a reason you added this?